site/en/docs/android/custom-tabs/integration-guide/index.md

```diff
@@ -48,29 +49,84 @@ customTabsIntent.launchUrl(this, Uri.parse(url));
 
--   `CustomTabColorScheme.navigationBarColor`
--   `CustomTabColorScheme.navigationBarDividerColor`
+Starting a new browser sessions is the recommended approach to launching a partial Custom Tag, as it enables you to [warm up the browser process](#warm-up-the-browser-process) and [pre-render webpages](/docs/android/custom-tabs/best-practices/#pre-render-content). If you don't need these features, using `startActivityForResult()` is the easier way to launch a partial custom tab.
 
-Custom Tabs will inherit the host app's color scheme for the UI properties above. You will be responsible for ensuring visual consistency for these properties before launching a Custom Tab.
+{% Aside 'gotchas' %}
+* Specifying the initial activity height will not have an effect if the default browser does not support resizing the Custom Tab. In this case, the intent extra will be ignored and the Custom Tab will span the complete display height.
 
-These new APIs are available from Chrome 107 and onwards.
+* Custom Tabs will inherit the host app's color scheme for the user interface properties above. You will be responsible for ensuring visual consistency for these properties before launching a Custom Tab. This means the `CustomTabColorScheme.navigationBarColor` and `CustomTabColorScheme.navigationBarDividerColor` properties do not work when building an intent for a Custom Tab.
+
+{% endAside %}
+
+
+### Launch a partial custom tab with an existing session
 
 ```java
-CustomTabsIntent customTabsIntent;
-...
-customTabsIntent = new CustomTabsIntent.Builder()
+CustomTabsSession customTabsSession;
+
+// ...
+
+CustomTabsIntent customTabsIntent = new CustomTabsIntent.Builder(session)
   .setInitialActivityHeightPx(500)
   .setCloseButtonPosition(CustomTabsIntent.CLOSE_BUTTON_POSITION_END)
-  ...
+  // ...
   .build();
+
+customTabsIntent.launchUrl(context, Uri.parse(url))
+```
+
+### Launch a partial custom tab via startActivityForResult
+
+```java
+private ActivityResultLauncher<String> mCustomTabLauncher = registerForActivityResult(new ActivityResultContract<String, Integer>() {
+    @Override
+    public Integer parseResult(int statusCode, @Nullable Intent intent) {
+        return statusCode;
+    }
+
+    @NonNull
+    @Override
+    public Intent createIntent(@NonNull Context context, String url) {
+        CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder()
+                .setInitialActivityHeightPx(500)
+                .setCloseButtonPosition(CustomTabsIntent.CLOSE_BUTTON_POSITION_END)
+                .setToolbarCornerRadiusDp(10);
+        Intent customTabsIntent = builder.build().intent;
+        customTabsIntent.setData(Uri.parse(url));
+        return customTabsIntent;
+    }
+}, new ActivityResultCallback<Integer>() {
+    @Override
+    public void onActivityResult(Integer statusCode) {
```

Comment:
Is it worth noting that Chrome is actually going to return a result?

Reply:
Absolutely, I've been wondering that as well, what is it going to return?